### PR TITLE
fix: Fix DependencyObjectGenerator for global namespaces and WPF head

### LINKED ISSUE #9423: Deriving DependencyObject under .NET 6/WPF errors with "does not implement .. 'DependencyObject.DispatcherQueue"
### Current behavior

This issue is suspiciously similar to #9380, but you only encounter this _after_ applying the recommended workaround to the WPF head:

```xml
  <Target Name="RemoveDuplicateAnalyzers" BeforeTargets="CoreCompile">
    <!-- Work around https://github.com/dotnet/wpf/issues/6792 -->

    <ItemGroup>
      <FilteredAnalyzer Include="@(Analyzer->Distinct())" />
      <Analyzer Remove="@(Analyzer)" />
      <Analyzer Include="@(FilteredAnalyzer)" />
    </ItemGroup>
  </Target>
```

After testing workarounds and gathering information for a few days, this doesn't seem to be related to the above issue. I can only assume there's something funky happening with the new Roslyn-based source generators in Uno.

Using the latest `dotnet new unoapp` template, if you create a class that derives `DependencyObject`:

```csharp
public sealed partial class NavigateToUriAction : DependencyObject, IAction
{
    public Uri NavigateUri
    {
        get => (Uri)GetValue(NavigateUriProperty);
        set => SetValue(NavigateUriProperty, value);
    }

    public static readonly DependencyProperty NavigateUriProperty = DependencyProperty.Register(
        nameof(NavigateUri),
        typeof(Uri),
        typeof(NavigateToUriAction),
        new PropertyMetadata(null));

    /// <inheritdoc/>
    public object Execute(object sender, object parameter)
    {
        if (NavigateUri != null)
        {
            _ = Windows.System.Launcher.LaunchUriAsync(NavigateUri);
        }
        else
        {
            throw new ArgumentNullException(nameof(NavigateUri));
        }

        return true;
    }
}
```

Then WPF head will fail to compile with the error:
```
 error CS0535: 'NavigateToUriAction' does not implement interface member 'DependencyObject.DispatcherQueue' 
```

If you manually add the code that should be getting generated (pulled from [here](https://github.com/unoplatform/uno/blob/master/src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs#L878)):
```cs
public global::Microsoft.UI.Dispatching.DispatcherQueue DispatcherQueue { get; } = global::Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
```

Then you get the error:
```
error CS0102: The type 'NavigateToUriAction' already contains a definition for 'DispatcherQueue' 
```

### Expected behavior

The WPF head should compile when valid code is used.

### How to reproduce it (as minimally and precisely as possible)

Repro project: [Test.zip](https://github.com/unoplatform/uno/files/9264106/Test.zip)

Note that only the WPF head has been set up for repro. Compile directly with MSBuild.

### Workaround

No known workaround. Tried forcing .NET to a version before the problematic Analyzer PR was introduced to .NET 6, but no available version of .NET 6 seems to work.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Latest default versions for `dotnet new unoapp` as of posting (4.4.13)

### Affected platforms

Skia (WPF)

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #9423

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs [hunk 1]
```diff
@@ -161,7 +161,7 @@ private void ProcessType(INamedTypeSymbol typeSymbol)
 					builder.AppendLineInvariant("using AppKit;");
 					builder.AppendLineInvariant("#endif");
 
-					using (builder.BlockInvariant($"namespace {typeSymbol.ContainingNamespace}"))
+					using (typeSymbol.ContainingNamespace.IsGlobalNamespace ? DisposableAction.NoOp : builder.BlockInvariant($"namespace {typeSymbol.ContainingNamespace}"))
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:164]:
```suggestion
					using (typeSymbol.ContainingNamespace.IsGlobalNamespace ? Uno.Actions.Null : builder.BlockInvariant($"namespace {typeSymbol.ContainingNamespace}"))
```

File: src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/DisposableAction.cs [hunk 2]
```diff
@@ -23,6 +23,8 @@ namespace Uno
 {
     internal class DisposableAction : IDisposable
     {
+		internal readonly static DisposableAction NoOp = new DisposableAction(() => { });
```
Comment (jeromelaban) [inline src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/DisposableAction.cs:26]:
```suggestion
```

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:164]: `Uno.Actions.Null` is not `IDisposable` so I don't think it works in `using`.

Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:164]: I can update the disposable action to `internal readonly static DisposableAction Null = new DisposableAction(Actions.Null);` though.

Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:164]: Right I'm sorry, that's not the right one, My mistake :) There's this one:
https://github.com/unoplatform/uno/blob/c94d459528296a91d0ac6cac89b3c78304f11b49/src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Disposables/NullDisposable.cs#L24

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**